modules/cnv_upload_virtctl.adoc

```diff
@@ -0,0 +1,48 @@
+// Module included in the following assemblies:
+//
+// * cnv_users_guide/cnv_users_guide.adoc
+
+[[upload-vmdisk-virtctl]]
+= Uploading a local disk image to a new PVC
+
+You can use `virtctl image-upload` to upload a virtual machine disk image from
+a client machine to your {product-title} cluster. This creates a PVC that can be
+associated with a virtual machine after the upload has completed.
+
+.Prerequisites
+
+* A virtual machine disk image, in RAW or QCOW2 format. It can be compressed
+using *xz* or *gzip*.
+* *kubevirt-virtctl* must be installed on the client machine.
+* The client machine must be xref:../install_config/router/default_haproxy_router.html#overview[configured]
+to trust the OpenShift router's certificate.
+
+.Procedure
+
+. Identify the following items:
+* File location of the VM disk image you want to upload
+* Name and size desired for the resulting PVC
+
+. Use the `virtctl image-upload` command to upload your VM image,
+making sure to include your chosen parameters. For example:
++
+----
+$ virtctl image-upload --uploadproxy-url=https://$(oc get route cdi-uploadproxy-route -n cdi -o=jsonpath='{.status.ingress[0].host}') --pvc-name=upload-fedora-pvc --pvc-size=10Gi --image-path=/images/fedora28.qcow2
```

Comment:
In an environment where CDI has been deployed through the HCO I see two differences:

- The upload route is in the `kubevirt-hyperconverged` namespace, not in the `cdi` namespace
- The route name is `cdi-uploadproxy`, not `cdi-uploadproxy-route`

So this command as-is wouldn't work on an environment deployed through HCO.

I believe that the idea in that case would be that image-upload would discover that through a `CDIConfig` —which isn't there in the env where I tested, but that's a different story— but if that's the case it should be noted.

Also, for readability: the command does get quite long. Maybe split it in multiple lines? And maybe separate the route discovery by creating a var and using it in the `virtctl` invocation?

Reply:
Oh, and another comment about the route: finding it (the `oc get route...` command) requires certain privileges, i.e. your user must be able to get routes in the cdi/hco namespace. Maybe this permission should be granted out of the box, but in my test env it wasn't. So it's either a bug or this should probably be documented here.

Reply:
We have automatic ingress detection so we can simplify this dramatically by removing the --uploadproxy-url parameter completely.